### ISSUE TITLE
[java]AvoidThrowingNullPointerException doesn't detect methods that throws an NullPointerException

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2533]
**Affects PMD Version:** 
6.22.0

**Rule:**
AvoidThrowingNullPointerException

**Description:**
AvoidThrowingNullPointerException doesn't detect methods that throws an NullPointerException
This rule is implemented through xpath search:
```
//AllocationExpression/ClassOrInterfaceType[@Image='NullPointerException']
```
This rule can detect <code>throw new NullPointerException();</code> but may forget to check methods.

**Code Sample demonstrating the issue:**
```
public foo() throw NullPointerException
```

**Expected outcome:**
false-negative

**Running PMD through:**
CLI

**Comment by zbynek** [comment]
There are two independent concepts that PMD checks:
* declaring which exceptions are thrown: [AvoidUncheckedExceptionsInSignatures](https://pmd.github.io/pmd/pmd_rules_java_design.html#avoiduncheckedexceptionsinsignatures) covers `NullPointerException` and other runtime exceptions
* actually throwing them: handled by [AvoidThrowingRawExceptionTypes](https://pmd.github.io/pmd/pmd_rules_java_design.html#avoidthrowingrawexceptiontypes) which replaces `AvoidThrowingNullPointerException`

You might want to enable both rules in your project.
